atomate/vasp/fireworks/exchange.py

```diff
@@ -0,0 +1,181 @@
+                mapping, detailed above.
+            name (str): Labels the FW.
+            c (dict): Config dict.
+            structures (list): Magnetic structures.
+            energies (list): Total energies of magnetic structures.
+
+        """
+
+        cutoff = heisenberg_settings["cutoff"]
+        tol = heisenberg_settings["tol"]
+        average = heisenberg_settings["average"]
+
+        fw_name = "%s %s" % (parent_structure.composition.reduced_formula, name)
+
+        additional_fields = {
+            "task_label": fw_name,
+            "exchange": {
+                "calc_type": name,
+                "wf_uuids": [],
+                "_source_wf_uuid": wf_uuid,
+            },
+        }
+
+        tasks = []
+
+        # Generate a HeisenbergModel with only <J> exchange
+        if average:
+            tasks.append(
+                HeisenbergModelMapping(
+                    db_file=db_file,
+                    wf_uuid=wf_uuid,
+                    cutoff=cutoff,
+                    tol=tol,
+                    average=average,
+                    structures=structures,
+                    energies=energies,
+                ))
+            tasks.append(
+                HeisenbergModelToDb(
+                    db_file=db_file,
+                    wf_uuid=wf_uuid,
+                    cutoff=cutoff)
+            )
+
+        else:
+            end_cutoff = 5.0  # Larger than this is unreasonable
+
+            for coff in np.linspace(cutoff, end_cutoff, 5):
+                tasks.append(
+                    HeisenbergModelMapping(
+                        db_file=db_file,
+                        wf_uuid=wf_uuid,
+                        cutoff=coff,
+                        tol=tol,
+                        average=average,
+                    ))
+                tasks.append(
+                    HeisenbergModelToDb(
+                        db_file=db_file,
+                        wf_uuid=wf_uuid,
+                        cutoff=cutoff)
+                )
+
+        super().__init__(tasks=tasks, name=fw_name, parents=parents)
+
+
+class VampireCallerFW(Firework):
+    def __init__(
+        self,
+        wf_uuid,
+        parent_structure,
+        parents,
+        db_file=DB_FILE,
+        mc_settings=None,
+        name="vampire caller",
+        average=True,
+    ):
+        """Run Vampire Monte Carlo from a HeisenbergModel.
+
+        * mc_settings:
+            mc_box_size (float): MC simulation box size in nm.
+            equil_timesteps (int): Number of MC equilibration moves.
+            mc_timesteps (int): Number of MC moves for averaging.
+
+        Args:
+            wf_uuid (int): Unique id for record keeping.
+            parent_structure (Structure): Magnetic ground state.
+            parents (FireWorks): Parent FWs.
+            db_file (str): Path to file containing db credentials.
+            mc_settings (dict): A configuration dict for monte carlo.
```

Comment:
I guess the type hinting for this should be `Optional[dict]`. And again, having this as option fits nicely in with the above!